src/sentry/tasks/assemble.py

```diff
@@ -179,26 +189,38 @@ def get_assemble_status(task, scope, checksum):
     notice or error message.
     """
     cache_key = _get_cache_key(task, scope, checksum)
-    rv = default_cache.get(cache_key)
+
+    if options.get("assemble.read_from_redis"):
+        client = _get_redis_cluster_for_assemble()
+        rv = client.get(cache_key)
+    else:
+        rv = default_cache.get(cache_key)
+
     if rv is None:
         return None, None
     return tuple(rv)
 
 
+@sentry_sdk.tracing.trace
 def set_assemble_status(task, scope, checksum, state, detail=None):
     """
     Updates the status of an assembling task. It is cached for 10 minutes.
     """
     cache_key = _get_cache_key(task, scope, checksum)
     default_cache.set(cache_key, (state, detail), 600)
+    redis_client = _get_redis_cluster_for_assemble()
+    redis_client.set(name=cache_key, value=str((state, detail)), ex=600)
```

Comment:
@asottile-sentry Is the `str()` correct for passing tuples to Redis?

Reply:
yeah this probably needs to be explicitly json encoded and decoded